docs/task-batching.adoc

```diff
@@ -0,0 +1,97 @@
+= Task Batching
+:base_version: 4.0.0
+:modules: processor
+
+== Introduction
+Task Batching is a feature to batch several tasks of type `T` to `List<T>` and process them at once.
+
+Decaton provides a *time-based* way and *size-based* way to batch multiple tasks.
+The *time-based* means that tasks in past linger time are processed on every linger time.
+The *size-based* means that tasks in past before reaching capacity are processed every time tasks’size reaches capacity.
+
+== When to use (Example)
+When downstream-DB supports batching I/O (which often very efficient)
+
+== Usage
+To use `Task Batching`, you only need to instantiate `BatchingProcessor`.
```

Comment:
Practically, developers will create their own class that inherits BatchingProcessor rather direct instantiation I guess.
So `you only need to implement BatchingProcessor` sounds more appropriate?

Reply:
The description and examples have been updated to use `InsertHelloTaskBatchingProcessor`, which inherits `BatchingProcessor<HelloTask>`.